src/gConsent/manage/getAccessGrantAll.ts

```diff
@@ -32,10 +32,35 @@ import type { IssueAccessRequestParameters } from "../type/IssueAccessRequestPar
 import { accessToResourceAccessModeArray } from "../util/accessToResourceAccessModeArray";
 import { getSessionFetch } from "../../common/util/getSessionFetch";
 import type { BaseGrantBody } from "../type/AccessVerifiableCredential";
+import { isAccessGrant } from "../guard/isAccessGrant";
+import { isBaseAccessGrantVerifiableCredential } from "../guard/isBaseAccessGrantVerifiableCredential";
+import { AccessGrant } from "../type/AccessGrant";
+import { getInherit, getResources } from "../../common/getters";
+
+// Iteratively build the list of ancestor containers from the breakdown of the
+// resource path: for resource https://pod.example/foo/bar/baz, we'll want the result
+// to be ["https://pod.example/", "https://pod.example/foo/", "https://pod.example/foo/bar/", https://pod.example/foo/bar/baz].
+const getAncestorUrls = (resourceUrl: URL) => {
```

Comment:
Shouldn't this be following the containment triples/link headers? Rather than being directory based?